docs/Toy/toy/rewrites/lower_toy_affine.py

```diff
@@ -411,7 +412,27 @@ def match_and_rewrite(self, op: toy.FuncOp, rewriter: PatternRewriter):
 class PrintOpLowering(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: toy.PrintOp, rewriter: PatternRewriter):
-        rewriter.replace_matched_op(PrintFormatOp("{}", op.input))
+        tensor_type = cast(toy.TensorTypeF64, op.input.type)
+        shape = tensor_type.get_shape()
+
+        format_str = "{}"
+
+        for dim in reversed(shape):
+            format_str = str([format_str] * dim).replace("'", "")
```

Comment:
Hmm, this feels a bit hacky, can't we instead do something like:

```py
format_str = "[" + ",".join([format_str] * dim) + "]"
```

I feel it would be easier to read the code and understand what it's doing that way?

Reply:
Super hacky :) Let me try your thing